web/src/main/java/com/garret/movies/controller/MovieOmdbController.java

```diff
@@ -0,0 +1,38 @@
+package com.garret.movies.controller;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.batch.core.*;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.repository.JobExecutionAlreadyRunningException;
+import org.springframework.batch.core.repository.JobInstanceAlreadyCompleteException;
+import org.springframework.batch.core.repository.JobRestartException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@AllArgsConstructor
+@Slf4j
+@RequestMapping("/api")
+public class MovieOmdbController {
+
+    private JobLauncher jobLauncher;
+    private Job job;
+
+    @GetMapping("/search/{title}")
+    public BatchStatus searchMovies(@PathVariable("title") String title)
+            throws JobParametersInvalidException, JobExecutionAlreadyRunningException,
+            JobRestartException, JobInstanceAlreadyCompleteException {
+
+        JobParametersBuilder parametersBuilder = new JobParametersBuilder();
+        parametersBuilder.addString("title", title);
+        parametersBuilder.addLong("time", System.currentTimeMillis());
+        JobParameters jobParameters = parametersBuilder.toJobParameters();
+        JobExecution jobExecution = jobLauncher.run(job, jobParameters);
+        log.info("Job execution status: " + jobExecution.getStatus());
+
+        return jobExecution.getStatus();
```

Comment:
@Garret-vin This implementations means synchronize job execution from HTTP call, so you will block call until your job will finish, its not right as you probably will have HTTP timeouts and in addition blocking thread always bad